### DISCUSSION TITLE
Windows should have a builtin console text editor

### DISCUSSION THREAD

**Discussion by driver1998** [microsoft/terminal#15651] [Ideas]
Once upon a time, there was good old `edit.com`, which served as a basic console mode text editor. It was a DOS app running on NTVDM but it got the job done.

But now, Windows does not have a console text editor builtin. We have a ton of choices of course, but what happen if you remote to a Windows Server via WinRM or SSH, and need to edit some config file or script?

It doesn't need to be too feature rich, even a notepad equivalent would be good. Or maybe there is some BSD/MIT opensource editors that can be intergrated like bsdtar and curl?

**Comment by driver1998** [comment]
An old post from OS/2 Museum which lists quite a few options: https://www.os2museum.com/wp/wanted-console-text-editor-for-windows/

**Comment by zadjii-msft** [comment]
@DHowett @crutkas This seems like a great idea. I wonder wtf ever happened to `edit`. Did no one ever port it to 64-bit, and we just... gave up on it?

**Reply by zadjii-msft** [reply]
We've discovered it's... very bad. `edit` is a `.COM` file. In `/base/mvdm/dos/v86/cmd/edit`, for future reference. That's... by all accounts, not getting resurrected any time soon.
